src/main/java/org/iot/dsa/dslink/java/v2/jdbc/DBConnectionNode.java

```diff
@@ -0,0 +1,346 @@
+        try {
+            res = new JDBCClosedTable(act, rSet, getLogger());
+        } catch (SQLException e) {
+            put(conn_status, DSString.valueOf(ConnStates.Unknown));
+            JDBCv2Helpers.cleanClose(rSet, null, null, getLogger());
+            throw new DSRequestException("Failed to retrieve data from database: " + e);
+        }
+        return res;
+    }
+
+    ResultSet executeQuery(String sqlQuery) {
+        Connection conn = null;
+        Statement stmt = null;
+        ResultSet rSet = null;
+        try {
+            conn = getConnection();
+            stmt = conn.createStatement();
+            connSuccess(true);
+            try {
+                rSet = stmt.executeQuery(sqlQuery);
+            } catch (SQLException e) {
+                put(conn_status, DSString.valueOf(ConnStates.Unknown));
+                JDBCv2Helpers.cleanClose(null, stmt, conn, getLogger());
+                throw new DSRequestException("Query failed: " + e);
+            }
+        } catch (SQLException e) {
+            connSuccess(false);
+            //noinspection ConstantConditions
+            JDBCv2Helpers.cleanClose(rSet, stmt, conn, getLogger());
+            warn("Failed to connect to Database: " + db_name.getValue(), e);
+            throw new DSRequestException("Database connection failed: " + e);
+        }
+        return rSet;
+    }
+
+    private DSAction makeUpdateAction() {
+        DSAction act = new DSAction() {
+            @Override
+            public ActionResult invoke(DSInfo info, ActionInvocation invocation) {
+                return ((DBConnectionNode) info.getParent()).runUpdate(invocation.getParameters());
+            }
+        };
+        act.addParameter(JDBCv2Helpers.QUERY, DSValueType.STRING, null);
+        return act;
+    }
+
+    private ActionResult runUpdate(DSMap params) {
+        String query = params.get(JDBCv2Helpers.QUERY).toString();
+        Connection conn = null;
+        Statement stmt = null;
+        try {
+            conn = getConnection();
+            stmt = conn.createStatement();
+            connSuccess(true);
+            try {
+                stmt.executeUpdate(query);
+            } catch (SQLException e) {
+                put(conn_status, DSString.valueOf(ConnStates.Unknown));
+                JDBCv2Helpers.cleanClose(null, stmt, conn, getLogger());
+                throw new DSRequestException("Update failed: " + e);
+            }
+        } catch (SQLException e) {
+            connSuccess(false);
+            //noinspection ConstantConditions
+            JDBCv2Helpers.cleanClose(null, stmt, conn, getLogger());
+            warn("Failed to connect to Database: " + db_name.getValue(), e);
+            throw new DSRequestException("Database connection failed: " + e);
+        }
+        return null;
+    }
+
+    abstract Connection getConnection() throws SQLException;
+
+    //TODO: Implement Streaming Queries
+/*    private DSAction makeStreamingQueryAction() {
```

Comment:
remove if not implemented/working.

Reply:
The code is there as a starting point to get that option working when we get around to it/as well as to help illustrate what the TODO item means by streaming queries. Maybe there is a more elegant way to store it. But making a new branch seemed too complicated.